File: backend/app/schemas/alert.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel


class AlertRuleCreate(BaseModel):
    device_id: Optional[int] = None
    sensor_id: Optional[int] = None
    parameter_name: str
    name: str
    condition: str = "gt"
    threshold_value: Optional[float] = None
    threshold_min: Optional[float] = None
    threshold_max: Optional[float] = None
    severity: str = "warning"
    category: str = "Inne"
    notify_channels: List[str] = []


class AlertRuleUpdate(BaseModel):
    name: Optional[str] = None
    enabled: Optional[bool] = None
    threshold_value: Optional[float] = None
    threshold_min: Optional[float] = None
    threshold_max: Optional[float] = None
    severity: Optional[str] = None
    category: Optional[str] = None
    notify_channels: Optional[List[str]] = None


class AlertRuleOut(BaseModel):
    id: int
    device_id: Optional[int] = None
    sensor_id: Optional[int] = None
    parameter_name: str
    name: str
    condition: str
    threshold_value: Optional[float] = None
    threshold_min: Optional[float] = None
    threshold_max: Optional[float] = None
    severity: str
    category: str
    enabled: bool
    notify_channels: List[str] = []
    created_at: datetime

    model_config = {"from_attributes": True}


class AlertEventOut(BaseModel):
    id: int
    rule_id: int
    device_id: Optional[int] = None
    sensor_id: Optional[int] = None
    value: Optional[float] = None
    severity: str
    category: str
    message: Optional[str] = None
    timestamp: datetime
    resolved_at: Optional[datetime] = None
    acknowledged: bool
    acknowledged_by: Optional[int] = None
    acknowledged_at: Optional[datetime] = None

    model_config = {"from_attributes": True}

File: frontend/src/types/alert.ts
export const ALERT_CATEGORIES = ['Temperatura', 'Komunikacja', 'Drzwi', 'Zasilanie', 'Sprzęt', 'Inne'] as const
export type AlertCategory = typeof ALERT_CATEGORIES[number]

export interface AlertRule {
  id: number
  device_id: number | null
  sensor_id: number | null
  parameter_name: string
  name: string
  condition: string
  threshold_value: number | null
  threshold_min: number | null
  threshold_max: number | null
  severity: 'info' | 'warning' | 'critical'
  category: string
  enabled: boolean
  notify_channels: string[]
  created_at: string
}

export interface AlertEvent {
  id: number
  rule_id: number
  device_id: number | null
  sensor_id: number | null
  value: number | null
  severity: 'info' | 'warning' | 'critical'
  category: string
  message: string | null
  timestamp: string
  resolved_at: string | null
  acknowledged: boolean
  acknowledged_by: number | null
  acknowledged_at: string | null
}
